examples2d/add_remove2.rs

```diff
@@ -19,7 +19,12 @@ pub fn init_world(testbed: &mut Testbed) {
         physics
             .colliders
             .insert(collider, handle, &mut physics.bodies);
-        graphics.add(window, handle, &physics.bodies, &physics.colliders);
+
+        if graphics.is_some() {
+            graphics
+                .unwrap()
+                .add(window.unwrap(), handle, &physics.bodies, &physics.colliders);
+        }
```

Comment:
Something like this should work:

```suggestion
        if let (Some(graphics), Some(window)) = (&mut graphics, &mut window) {
            graphics.add(*window, handle, &physics.bodies, &physics.colliders);
        }
```
Matching on `&mut Option` will make sure that you don't move the `&mut T` out of `Option<&mut T>`.

Reply:
Right,

I think I get this. Regardless, I have changed all the callback related example code to reflect these suggestions.

Thank you, this saved me hours of hair pulling.

Thanks